integrations/nifi/nifi-nar-bundles/nifi-spark-bundle/nifi-spark-processors/src/main/java/com/thinkbiganalytics/nifi/v2/spark/ExecuteSparkJob.java

```diff
@@ -579,9 +515,61 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         }
     }
 
+    private Map<String, String> getDatasources(ProcessSession session, FlowFile flowFile, String PROVENANCE_JOB_STATUS_KEY, String datasourceIds, MetadataProviderService metadataService, List<String> extraJarPaths) throws JsonProcessingException {
+        final Map<String, String> env = new HashMap<>();
+
+        if (StringUtils.isNotBlank(datasourceIds)) {
+            final StringBuilder datasources = new StringBuilder(10240);
+            final ObjectMapper objectMapper = new ObjectMapper();
+            final MetadataProvider provider = metadataService.getProvider();
+
+            for (final String id : datasourceIds.split(",")) {
+                datasources.append((datasources.length() == 0) ? '[' : ',');
+
+                final Optional<Datasource> datasource = provider.getDatasource(id);
+                if (datasource.isPresent()) {
+                    if (datasource.get() instanceof JdbcDatasource && StringUtils.isNotBlank(((JdbcDatasource) datasource.get()).getDatabaseDriverLocation())) {
+                        final String[] databaseDriverLocations = ((JdbcDatasource) datasource.get()).getDatabaseDriverLocation().split(",");
+                        extraJarPaths.addAll(Arrays.asList(databaseDriverLocations));
+                    }
+                    datasources.append(objectMapper.writeValueAsString(datasource.get()));
+
+                } else {
+                    getLog().error("Required datasource {} is missing for Spark job: {}", new Object[]{id, flowFile});
+                    flowFile = session.putAttribute(flowFile, PROVENANCE_JOB_STATUS_KEY, "Invalid data source: " + id);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "flowFile". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1226)